port/blueprint-permissions/resource_test.go

```diff
@@ -0,0 +1,510 @@
+					resource.TestCheckResourceAttr("port_blueprint.microservice", "icon", "Terraform"),
+				),
+			},
+			{
+				Config: testAccPortBlueprintResourceBasic + testAccBaseBlueprintPermissionsConfigUpdate,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("port_blueprint.microservice", "identifier", blueprintIdentifier),
+					resource.TestCheckResourceAttr("port_blueprint.microservice", "title", "TF Provider Test"),
+					resource.TestCheckResourceAttr("port_blueprint.microservice", "icon", "Terraform"),
+					resource.TestCheckResourceAttr("port_blueprint_permissions.microservice_permissions", "blueprint_identifier", blueprintIdentifier),
+					resource.TestCheckResourceAttr("port_blueprint_permissions.microservice_permissions", "id", blueprintIdentifier),
+					resource.TestCheckResourceAttr("port_blueprint_permissions.microservice_permissions", "entities.register.roles.#", "1"),
+					resource.TestCheckResourceAttr("port_blueprint_permissions.microservice_permissions", "entities.register.roles.0", "Member"),
+					resource.TestCheckResourceAttr("port_blueprint_permissions.microservice_permissions", "entities.register.users.#", "0"),
+					resource.TestCheckResourceAttr("port_blueprint_permissions.microservice_permissions", "entities.register.teams.#", "0"),
+					resource.TestCheckResourceAttr("port_blueprint_permissions.microservice_permissions", "entities.update_properties.myStringIdentifier.roles.#", "1"),
+					resource.TestCheckResourceAttr("port_blueprint_permissions.microservice_permissions", "entities.update_properties.myStringIdentifier.roles.0", "Member"),
+				)},
+		},
+	})
+}
+
+func TestAccPortBlueprintPermissionsWithRelations(t *testing.T) {
+	blueprintMicroserviceIdentifier := utils.GenID()
+	blueprintEnvIdentifier := utils.GenID()
+	err := os.Setenv("PORT_BETA_FEATURES_ENABLED", "true")
+	if err != nil {
+		t.Fatal(err)
+	}
+	var testAccPortBlueprintResourceBasic = fmt.Sprintf(`
+resource "port_blueprint" "environment" {
+  title      = "Environment"
+  icon       = "Environment"
+  identifier = "%s"
+  properties = {
+    string_props = {
+      "name" = {
+        type  = "string"
+        title = "name"
+      }
+      "docs-url" = {
+        title  = "Docs URL"
+        format = "url"
+      }
+    }
+  }
+}
+
+
+resource "port_blueprint" "microservice" {
+  identifier            = "%s"
+  title                 = "TF Provider Test"
+  icon                  = "Terraform"
+  description			= ""
+  relations = {
+    "environment" = {
+      title    = "Test Relation"
+      required = "true"
+      target   = port_blueprint.environment.identifier
+    }
+  }
+}
+`, blueprintEnvIdentifier, blueprintMicroserviceIdentifier)
+
+	teamName := utils.GenID()
+	var testAccBaseBlueprintPermissionsConfigUpdate = fmt.Sprintf(`
+
+	resource "port_team" "team" {
+		name = "%s"
+		description = "Test description"
+		users = []
+	}
```

Comment:
same here, though I would like to have at least one test checking that the teams field works

Reply:
Added